pkgs/by-name/ku/kube-state-metrics/package.nix

```diff
@@ -0,0 +1,34 @@
+{
+  stdenv,
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule rec {
+  pname = "kube-state-metrics";
+
+  version = "2.12.0";
+  vendorHash = "sha256-A/oiFEWM3TOtk4V3xpehJ+5oKSyzYKZHVlp3wjykFRA=";
+
+  src = fetchFromGitHub {
+    owner = "kubernetes";
+    repo = "kube-state-metrics";
+    rev = "v${version}";
+    sha256 = "sha256-EBSlufgzT4zgxTO5B0/RHHJJ2GgtU3U9ypd1QjC3knE=";
+  };
+
+  excludedPackages = [ "./tools" ];
+
+  preCheck = ''
+    rm -r tests/e2e
```

Comment:
Could you do `doCheck = false;` instead if there are no other tests (had a short look) or open an issue to provide a command to run only the not e2e-tests and change the command in checkPhase to that if solved upstream?

Reply:
Thanks @SebTM - so like this? Did check https://github.com/kubernetes/kube-state-metrics/tree/main/tests/e2e where there seems to be several small tests based on some kind. But yes it does not kind of work in our NixOS environment. Maybe I will later try to contact upstream. But for now this work for us. We have used kube-state-metrics for > 4 years now so about time time to upstream :-).